pkg/state/finalization.go

```diff
@@ -0,0 +1,85 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/pkg/errors"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+)
+
+const finalizationKey = "finalization"
+
+var ErrNoFinalization = errors.New("no finalized blocks recorded")
+var ErrNoFinalizationHistory = errors.New("no finalization in history")
+
+type finalizationItem struct {
+	Block                proto.BlockHeader `cbor:"0,keyasint,omitempty"`
```

Comment:
No need to store block header second time here, can be retrieved by height or blockID.

Reply:
Fixed